src/codegen/constant_translator.cpp

```diff
@@ -0,0 +1,108 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// constant_translator.cpp
+//
+// Identification: src/codegen/constant_translator.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/constant_translator.h"
+
+#include "expression/constant_value_expression.h"
+#include "type/value_peeker.h"
+#include "codegen/value_peeker_proxy.h"
+#include "codegen/value_proxy.h"
+#include "codegen/parameter.h"
+
+namespace peloton {
+namespace codegen {
+
+// Constructor
+ConstantTranslator::ConstantTranslator(
+    const expression::ConstantValueExpression &exp, CompilationContext &ctx)
+    : ExpressionTranslator(exp, ctx), ctx_(ctx) {
```

Comment:
Again, I was trying to help. Meticulously writing these same comments may prevent you from missing a case. (though it turns out to be perfectly fine now)